docs/resources/node_template.md

```diff
@@ -235,12 +235,12 @@ The following attributes are exported:
 * `auth_url` - (Required) OpenStack authentication URL (string)
 * `availability_zone` - (Required) OpenStack availability zone (string)
 * `region` - (Required) OpenStack region name (string)
-* `username` - (Required) OpenStack username (string)
+* `username` - (Required**) OpenStack username (string)
 * `active_timeout`- (Optional) OpenStack active timeout Default `200` (string)
 * `cacert` - (Optional) CA certificate bundle to verify against (string)
 * `config_drive` - (Optional) Enables the OpenStack config drive for the instance. Default `false` (bool)
 * `domain_id` - (Required*) OpenStack domain ID. Identity v3 only. Conflicts with `domain_name` (string)
-* `domain_name` - (Required*) OpenStack domain name. Identity v3 only. Conflicts with `domain_id` (string)
+* `domain_name` - (Required**) OpenStack domain name. Identity v3 only. Conflicts with `domain_id` (string)
```

Comment:
What is the reason of `domain_name` argument modification??

Reply:
domain name and id are no longer required when using application credentials.

Reply:
just checked the documentation again (https://docs.openstack.org/keystone/latest/user/application_credentials.html#using-application-credentials). They're not required if using `application_credential_id`, also for the username. I fixed the note below.